test/middleware/request.js

```diff
@@ -1,98 +1,118 @@
 import test from 'ava';
 import nock from 'nock';
+import sinon from 'sinon';
+import rewire from 'rewire';
 
 import { CALL_API } from '../../app/src/js/actions/types';
-import { requestMiddleware } from '../../app/src/js/middleware/request';
+const request = rewire('../../app/src/js/middleware/request');
 
 const port = process.env.FAKEAPIPORT || 5001;
+const token = 'fake-token';
 
-test.beforeEach((t) => {
-  const doDispatch = () => {};
-  const doGetState = () => ({
-    api: {
-      tokens: {
-        token: 'fake-token'
-      }
+const requestMiddleware = request.__get__('requestMiddleware');
+const loginErrorStub = sinon.stub();
+request.__set__('loginError', () => {
+  loginErrorStub();
+});
+
+const getStateStub = sinon.stub().callsFake(() => ({
+  api: {
+    tokens: {
+      token
     }
-  });
-  t.context.nextHandler = requestMiddleware({dispatch: doDispatch, getState: doGetState});
+  }
+}));
+const dispatchStub = sinon.stub();
+const nextStub = sinon.stub();
+
+const create = () => {
+  const store = {
+    getState: getStateStub,
+    dispatch: dispatchStub
+  };
+  const next = nextStub;
+
+  const invokeMiddleware = action => requestMiddleware(store)(next)(action);
 
+  return { store, next, invokeMiddleware };
+};
+
+test.beforeEach((t) => {
   t.context.defaultConfig = {
     json: true,
     resolveWithFullResponse: true,
     simple: false
   };
 
   t.context.expectedHeaders = {
-    Authorization: 'Bearer fake-token',
+    Authorization: `Bearer ${token}`,
     'Content-Type': 'application/json'
   };
 });
 
-test('should pass action to next if not an API request action', (t) => {
+test.afterEach.always(() => {
+  nock.cleanAll();
+  dispatchStub.resetHistory();
+  getStateStub.resetHistory();
+  nextStub.resetHistory();
+  loginErrorStub.resetHistory();
+});
+
+test.serial('should pass action to next if not an API request action', (t) => {
   const actionObj = {};
```

Comment:
```suggestion
const actionObj = { type: 'ANYTHING', some: 'action' };```
Can we add something just to make sure we're not just dumping out an empty object? (I know we aren't)

Reply:
don't commit that suggestion, my comment was included somehow.